torch/csrc/utils/python_dispatch.cpp

```diff
@@ -50,6 +54,70 @@ inline torch::CppFunction dispatch_str(const char* key, Func&& raw_f) {
   }
 }
 
+class PythonKernelHolder : public c10::OperatorKernel {
+  SafePyObject* func_;
+public:
+  PythonKernelHolder(py::object func) : func_(new SafePyObject(func.release().ptr(), getPyInterpreter())) {}
+  // This is a generally useful pattern and safer than directly using pybind11's
+  // py::object destructor.  This is because this object may outlive
+  // libtorch_python, so we want to disarm the deallocation if that happens.
+  // PyInterpreter does this correctly, pybind11 does not.
+  ~PythonKernelHolder() {
+    getPyInterpreter()->decref(func_->ptr(getPyInterpreter()), /*is_tensor*/ false);
```

Comment:
You shouldn't need this anymore; SafePyObject should be stored directly by value and its destructor will take care of this logic for you. (This comment is mooted by the one above, but I feel also obligated to mention: this current code leaks SafePyObject as every new has to be paired with a delete.)